blockchain/blockdao/blockdao.go

```diff
@@ -117,30 +119,33 @@ type (
 	}
 
 	blockDAO struct {
-		compressBlock bool
-		kvStore       db.KVStore
-		indexer       BlockIndexer
-		htf           db.RangeIndex
-		kvStores      sync.Map //store like map[index]db.KVStore,index from 1...N
-		topIndex      atomic.Value
-		timerFactory  *prometheustimer.TimerFactory
-		lifecycle     lifecycle.Lifecycle
-		headerCache   *cache.ThreadSafeLruCache
-		bodyCache     *cache.ThreadSafeLruCache
-		footerCache   *cache.ThreadSafeLruCache
-		cfg           config.DB
-		mutex         sync.RWMutex // for create new db file
-		tipHeight     uint64
+		compressBlock    bool
+		kvStore          db.KVStore
+		indexer          BlockIndexer
+		systemLogIndexer systemlog.Indexer
```

Comment:
Don't add this variable.
indexer could be a wrapper of a list of indexers, no need to modify blockdao.go at all

Reply:
I consider this before, but the problem is it would affect many files. I will do it now.